src/cargo/ops/cargo_package.rs

```diff
@@ -743,10 +744,12 @@ fn check_repo_state(
                         .and_then(|p| p.to_str())
                         .unwrap_or("")
                         .replace("\\", "/");
-                    return Ok(Some(VcsInfo {
-                        git: git(p, src_files, &repo, &opts)?,
-                        path_in_vcs,
-                    }));
+                    let git = git(p, src_files, &repo, &opts)?;
```

Comment:
Could `fn git()` return a `None` when nothing needed to be recorded? So that we don't need `skip_serializing_if`  for `sha1` field.

Reply:
Agree with you and updated it